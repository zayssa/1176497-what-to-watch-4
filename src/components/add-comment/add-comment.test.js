import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import AddComment from "./add-comment.jsx";
import history from "../../history";


it(`Render AddComment page`, () => {
  const tree = renderer
    .create(<Router history={history}><AddComment
      films={[{
        id: 1,
        title: `Look`,
        poster: `img/avatar.jpg`,
        genre: `Comedy`,
        year: 2002,
        rating: 7.6,
        count: 178,
        bg: `img/bg-the-grand-budapest-hotel.jpg`,
        text: `In the 1930s, the Grand Budapest Hotel is a popular
        European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).
        Zero, a junior lobby boy, becomes Gustave&apos;s friend and protege.
        Gustave prides himself on providing first-class service to the hotel&apos;s guests,
        including satisfying the sexual needs of the many elderly women who stay there.
        When one of Gustave&apos;s lovers dies mysteriously, Gustave finds himself
        the recipient of a priceless painting and the chief suspect in her murder.`,
        director: `Billy Zane`,
        actors: [`Sherilyn Fenn`, `Michelle Rodriguez`, `Patrick Swayze`, `Benedict Cumberbatch`, `Emily Blunt`],
        preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        runtime: 102,
        previewImage: `img/the-grand-budapest-hotel.jpg`,
        bgColor: `#cccccc`,
        videoLink: `https://some-link`,
        isFavorite: true
      }]}
      match={{
        params: {
          filmId: `1`
        }
      }}
      userInfo={{
        "id": 1,
        "email": `qwe@mail.ru`,
        "name": `123456`,
        "avatar_url": `img/avatar.png`
      }} /></Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
