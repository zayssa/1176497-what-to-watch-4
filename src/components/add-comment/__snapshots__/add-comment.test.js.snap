// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AddComment page 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <symbol
        id="add"
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            id="+"
            points="10.777832 11.2880859 10.777832 19.5527344 8.41650391 19.5527344 8.41650391 11.2880859 0.627929688 11.2880859 0.627929688 8.92675781 8.41650391 8.92675781 8.41650391 0.662109375 10.777832 0.662109375 10.777832 8.92675781 18.5664062 8.92675781 18.5664062 11.2880859"
          />
        </g>
      </symbol>
      <symbol
        id="full-screen"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M0 3.14286L0 11H3.14286L3.14286 3.14286L11 3.14286V0H3.14286H0L0 3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="in-list"
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="pause"
        viewBox="0 0 14 21"
      >
        <title>
          Artboard
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Artboard"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="0 -1.11910481e-13 4 -1.11910481e-13 4 21 0 21"
          />
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="10 -1.11910481e-13 14 -1.11910481e-13 14 21 10 21"
          />
        </g>
      </symbol>
    </svg>
  </div>,
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__header"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Look"
          src="img/avatar.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <nav
          className="breadcrumbs"
        >
          <ul
            className="breadcrumbs__list"
          >
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
                href="/films/1"
                onClick={[Function]}
              >
                Look
              </a>
            </li>
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
              >
                Add review
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <a
              href="/mylist"
              onClick={[Function]}
            >
              <img
                alt="User avatar"
                height="63"
                src="https://4.react.pages.academyimg/avatar.png"
                width="63"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="Look"
          height="327"
          src="img/avatar.jpg"
          width="218"
        />
      </div>
    </div>
    <div
      className="add-review"
    >
      <form
        action="#"
        className="add-review__form"
        onSubmit={[Function]}
      >
        <div
          className="rating"
        >
          <div
            className="rating__stars"
          >
            <input
              className="rating__input"
              id="star-1"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="1"
            />
            <label
              className="rating__label"
              htmlFor="star-1"
            >
              Rating 1
            </label>
            <input
              className="rating__input"
              id="star-2"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="2"
            />
            <label
              className="rating__label"
              htmlFor="star-2"
            >
              Rating 2
            </label>
            <input
              className="rating__input"
              id="star-3"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="3"
            />
            <label
              className="rating__label"
              htmlFor="star-3"
            >
              Rating 3
            </label>
            <input
              className="rating__input"
              id="star-4"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="4"
            />
            <label
              className="rating__label"
              htmlFor="star-4"
            >
              Rating 4
            </label>
            <input
              className="rating__input"
              defaultChecked={true}
              id="star-5"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="5"
            />
            <label
              className="rating__label"
              htmlFor="star-5"
            >
              Rating 5
            </label>
          </div>
        </div>
        <div
          className="add-review__text"
        >
          <textarea
            className="add-review__textarea"
            id="review-text"
            name="review-text"
            onChange={[Function]}
            placeholder="Review text"
          />
          <div
            className="add-review__submit"
          >
            <button
              className="add-review__btn"
              disabled={true}
              style={
                Object {
                  "cursor": "default",
                  "opacity": 0.5,
                }
              }
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>,
]
`;
